woocommerce/woocommerce
# Product gallery/fix console error

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/52936

This PR reverts the mismatch error that is shown when Product Gallery block is added and saved/refreshed. This change was first implemented [here](https://github.com/woocommerce/woocommerce/pull/51189) to fix two issues [here](https://github.com/woocommerce/woocommerce/issues/51154) and [here](https://github.com/woocommerce/woocommerce/issues/51175) but caused a regression.

- While I reverted the mismatch error trap, I kept the defensive guards to prevent https://github.com/woocommerce/woocommerce/issues/51154 and https://github.com/woocommerce/woocommerce/issues/51175 from happening.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

### Test Case 1

1. Go to Appearance->Editor->Templates->WooCommerce->Single Product Template.
2. Click on the Product Image Gallery block in the Document Overview (list view) and remove it.
3. In it's place, add the Product Gallery (Beta) block. Ensure there are no console errors.
4. Save the template and ensure you see the block in the frontend and functioning as expected.

### Test Case 2

1. Go to Appearance->Editor->Patterns->Add New Pattern.
2. Enter any name for your pattern and click on Add.
3. Click on the Document Overview (list view) and add the Product Gallery (Beta) block to the page. Save.
4. Go to Appearance->Editor->Templates->WooCommerce->Single Product Template.
5. Click on the Product Image Gallery block in the Document Overview (list view) and remove it.
6. In it's place, add the new pattern you just added in step 2.
7. Ensure there are no errors or hanging in the browser.
8. Save the template and ensure you see the block in the frontend and functioning as expected.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Product Gallery (Beta) - Fix block not rendering after being added.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
